builtin/string.go

```diff
@@ -346,11 +348,22 @@ var _ = method(string_NumberQ, "()")
 
 func string_NumberQ(this Value) Value {
 	// see also Lexer.number
-	return SuBool(numberPat.Matches(ToStr(this)))
+	lexer := lexer.NewLexer(ToStr(this))
+	item := lexer.Next()
+	if item.Token == tok.Add || item.Token == tok.Sub {
+		item = lexer.Next()
+	}
+	if item.Token == tok.Number {
+		item = lexer.Next()
+		if item.Token == tok.Eof {
+			return True
+		}
+	}
+	return False
 }
 
 // benchmark shows Suneido regex is faster than Go regexp for this
-var numberPat = regex.Compile(`\A[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d\d?)?\Z`)
+var numberPat = regex.Compile(`\A[+-]?(\d+([_\d]+)*([\.]([_\d]+)*)?|([\.]\d+([_\d]+)*))([eE][+-]?\d+)?\Z`)
```

Comment:
We can delete the comment and numberPat.

Reply:
yes, I still haven't changed the tests. I'll do it with that push